### PR TITLE
Update Rules NodeJS to v5

### PR DESCRIPTION
**Pull request by devversion**
Continuation of #45235

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -103,20 +108,8 @@ executors:
...
     resource_class: windows.2xlarge
-    # CircleCI windows VMs do have the GitBash shell available:
-    # https://github.com/CircleCI-Public/windows-preview-docs#shells
-    # But in this specific case we really should not use it because Bazel must not be ran from
-    # GitBash. These issues discuss why:
-    # https://github.com/bazelbuild/bazel/issues/5751
-    # https://github.com/bazelbuild/bazel/issues/5724#issuecomment-410194038
-    # https://github.com/bazelbuild/bazel/issues/6339#issuecomment-441600879
-    shell: powershell.exe -ExecutionPolicy Bypass
+    shell: bash
```
Comment (gkalpak) [inline .circleci/config.yml:111]: OOC, why this change?

File: .circleci/config.yml [hunk 2]
```diff
@@ -724,15 +699,45 @@ jobs:
...
+      # Windows might error when `bazel` is invoked from the project node modules. The Bazel
+      # invocation might modify the symlinked project `node_modules` again, causing failures.
+      - yarn_install
+      - run: yarn global add @bazel/bazelisk@${BAZELISK_VERSION}
+
       - run:
           name: Build all windows CI targets
-          command: bazel build //packages/compiler-cli/...
+          command: |
+            $(yarn global bin)/bazelisk build //packages/compiler-cli/...
```
Comment (gkalpak) [inline .circleci/config.yml:727]: Why is `$(yarn global bin)` part necessary? Is it because the `PATH` is not updated to include the globally installed yarn modules?

**Review by gkalpak — APPROVED** [review]
:tada:

Thx for getting this sorted out, @devversion 💯 🥇 🏆
I know how difficult this has been (esp. without a Windows machine at hand) 🙇

(BTW, there is a typo in the last commit: `Shortns` --> `Shortens`)

File: .circleci/config.yml (hunk 1 above)
Comment (devversion) [reply .circleci/config.yml:111]: Because it allows us to share most setup code, and it's usually how Bazel is invoked on Windows (with Git Bash). A lot of transitive Bazel code relies on it, especially in our setup so this is a pre-requisite

File: .circleci/config.yml (hunk 2 above)
Comment (devversion) [reply .circleci/config.yml:727]: Correct. The Yarn global folder is not part of the `PATH` (previously it was with the special `ps1` setup) but it's not worth keeping that for the two cases here.

File: .circleci/config.yml [hunk 3]
```diff
@@ -724,15 +699,45 @@ jobs:
...
-      - setup_win
+      - checkout
+      - save_month_to_file
+      - init_environment
+
+      # Setup the proper NodeJS version. The image comes pre-installed with `nvm-windows`.
+      - run:
+          name: Install NodeJS version from `.nvmrc`.
+          command: |
+            nvm install $(cat .nvmrc)
```
Comment (josephperrott) [inline .circleci/config.yml:710]: Running `nvm install` will also set it as the version to currently use, so we shouldn't need `nvm use` as well

Additionally, if you run `nvm install` (or `nvm use`) it should just use the nearest `.nvmrc` file correct?

**Review by josephperrott — APPROVED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (devversion) [reply .circleci/config.yml:710]: Not here unfortunately since this is the Windows-based `nvm` 😄

Comment (josephperrott) [reply .circleci/config.yml:710]: womp womp

**Review by josephperrott — APPROVED** [review]
LGTM

Reviewed-for: global-approvers

**Comment by dylhunn** [comment]
Merging into 14, Joey will send a patch port.
